CHANGES

```diff
@@ -9,6 +9,11 @@
 
  Changes between 1.1.1 and 3.0.0 [xx XXX xxxx]
 
+  *) The security strength of SHA1 and MD5 based signatures has been reduced.
+     This results in SSL 3, TLS 1.0, TLS 1.1 and DTLS 1.0 no longer working at
```

Comment:
It also results in X509 certs not verifying outside of TLS if they are signed using SHA1?

Reply:
And if so, is there actually a workaround to reenable if you want to verify a SHA1 signed X.509 outside of TLS? Because for TLS you do it via the cipher string.

Reply:
IIRC, when verifying X.509 certificate chains outside of TLS, the default security level is 0. To get some other security level and "apps" user would have to pass an explicit "-auth_level" option, while an application would have to invoke an appropriate X509_VERIFY_PARAM call to set a non-zero security level.

This means we're changing the meaning of level 1 for applications that choose to apply level 1, but outside TLS the default level is IIRC still 0.

Reply:
Just checked, as the default security level is "-1" (not set), and with that certificate checks don't check the algorithm security levels.  See:
https://github.com/openssl/openssl/blob/master/crypto/x509/x509_vpm.c#L468-L530

Reply:
No, the X509 related changes are in #10786, this PR is only about the signature in SSL/TLS. I've updated the commit and CHANGES entry to be more clear about it.

Reply:
So this PR looks reasonable, and seems to do what it claims to do.  What remains I guess is to carefully adjust the tests until they pass, making sure that all the adjustments are what we expect, and don't just sweep any problems under the rug...